test/tokenizer/index.spec.ts

```diff
@@ -376,6 +376,262 @@ describe('scan', () => {
           },
         ].forEach(({ actual, expected }) => expect(actual).to.eql(expected));
       });
+
+      describe('custom parameters', () => {
+        it('should allow positional parameters for all dialects', () => {
+          const paramTypes: ParamTypes = {
+            positional: true,
+          };
+
+          const expected = {
+            type: 'parameter',
+            value: '?',
+            start: 0,
+            end: 0,
+          };
+
+          (
+            ['mssql', 'psql', 'oracle', 'bigquery', 'sqlite', 'mysql', 'generic'] as Array<Dialect>
+          ).forEach((dialect) => {
+            [
+              {
+                actual: scanToken(initState('?'), dialect, paramTypes),
+                expected,
+              },
+            ].forEach(({ actual, expected }) => expect(actual).to.eql(expected));
+          });
+        });
+
+        it('should allow numeric parameters for all dialects', () => {
+          const paramTypes: ParamTypes = {
+            numbered: ['$', '?', ':'],
+          };
+
+          const expected = [
+            {
+              type: 'parameter',
+              value: '$1',
+              start: 0,
+              end: 1,
+            },
+            {
+              type: 'parameter',
+              value: '?1',
+              start: 0,
+              end: 1,
+            },
+            {
+              type: 'parameter',
+              value: ':1',
+              start: 0,
+              end: 1,
+            },
+            {
+              type: 'parameter',
+              value: 'unknown',
```

Comment:
This doesn't seem right to me, where if it's not a parameter as it's not a number in this case, we shouldn't be classifying it as such.

Reply:
How do we want to classify it? I believe this was in the old code

Reply:
I think it would make more sense to be `type: 'unknown'` if we don't think it's a variable because the given dialect does not support that variable format, and the value should _always_ be the actual token value.